src/main.rs

```diff
@@ -548,4 +549,47 @@ mod tests {
         let names: Vec<_> = group.iter_mut().map(|s| s.name.as_str()).collect();
         assert_eq!(vec!["boo", "bar"], names);
     }
+
+    #[test]
+    fn check_env_config_without_clear() {
+        set_var("GLOBAL_VAR_123", "present");
+
+        let yaml = "
+            name: bar
+            key: b
+            env:
+              FOO: bar
+            cmd: echo -n \"The value of FOO is $FOO and GLOBAL_VAR_123 is $GLOBAL_VAR_123\"
+        ";
+
+        let task: Task = serde_yaml::from_str(yaml).unwrap();
+
+        let output = create_process(&task, false).unwrap().wait_with_output().unwrap();
+
+        remove_var("GLOBAL_VAR_123");
+
+        assert_eq!("The value of FOO is bar and GLOBAL_VAR_123 is present", String::from_utf8_lossy(&output.stdout));
+    }
+
+    #[test]
+    fn check_env_config_with_clear() {
+        set_var("GLOBAL_VAR_234", "global");
+
+        let yaml = "
+            name: bar
+            key: b
+            env:
+              FOO: bar
+            clear_env: true
+            cmd: echo -n \"The value of FOO is $FOO and GLOBAL_VAR_234 is $GLOBAL_VAR_234\"
+        ";
+
+        let task: Task = serde_yaml::from_str(yaml).unwrap();
+
+        let output = create_process(&task, false).unwrap().wait_with_output().unwrap();
+
+        remove_var("GLOBAL_VAR_234");
```

Comment:
I think this is a totally justified case for using global env-variables. The only thing I would recommend doing is to leverage Rust `drop` facilities to get rid of explicit env-var clear. What you can do instead is to create `EnvVar` with a custom `Drop` that will clean up the env variable when leaving the lexical scope. It will clean up even if `wait_with_output().unwrap()` or `from_str(...).unwrap()` will fire.

```rust
fn session_env_var(name: impl Into<String>, value: impl Into<String>) -> EnvVar {
    let name = name.into();
    let value = value.into();
    // set env
    EnvVar { name, value }
}

struct EnvVar {
    name: String,
    value: String,
}

impl Drop for EnvVar {
    fn drop(&mut self) {
        // clean env
    }
}
```